Assets/Scripts/InventoryAndItemsScripts/Items/Interactables/UIInteractable.cs

```diff
@@ -8,25 +8,31 @@ namespace Items.Interactables
     //This class should eventually be updated to better match how the rest of the UI is set up
     public class UIInteractable : MonoBehaviour, IInteractable
     {
-        [SerializeField] private GameObject UIPrefab;
+        [SerializeField] protected GameObject UIPrefab;
+        [SerializeField] private GameObject player;
 
-        private bool isActive = false;
-        private GameObject instantiatedUIPrefab;
-        private InteractableOverlayHelper interactableOverlay;
+        protected bool isActive = false;
+        protected GameObject instantiatedUIPrefab;
+        protected InteractableOverlayHelper interactableOverlay;
 
         public bool DestroyAfterInteracting { get { return false; } }
 
         void Awake()
+        {
+            GetInteractableOverlayComponent();
+
+            //Creates prefab in the center
+            this.instantiatedUIPrefab = Instantiate(this.UIPrefab, new Vector3(0, 0, 0), Quaternion.identity);
+            this.instantiatedUIPrefab.SetActive(false);
+        }
+
+        protected void GetInteractableOverlayComponent()
```

Comment:
whats the reason here for changing these to protected? i don't see any classes that inherit from it

Reply:
i'm blind and i just saw it, this isn't best practice but i think it's fine for our project